File: domain/src/main/java/com/everis/domain/interactor/GetIntroMessage.java
package com.everis.domain.interactor;

import com.everis.domain.executor.PostExecutorThread;
import com.everis.domain.repository.DatabaseRepository;

import rx.Observable;

/**
 * Created by everis on 26/04/18.
 */

public class GetIntroMessage extends UseCase {

    private DatabaseRepository repository;

    public GetIntroMessage(PostExecutorThread postExecutorThread, DatabaseRepository repository) {
        super(postExecutorThread);
        this.repository = repository;
    }

    @Override
    protected Observable buildUseCaseObservable() {
        return repository.getIntroMessage();
    }
}

File: domain/src/main/java/com/everis/domain/interactor/GetIntroMessage.kt
package com.everis.domain.interactor

import com.everis.domain.executor.PostExecutorThread
import com.everis.domain.model.IntroMessage
import com.everis.domain.repository.DatabaseRepository

import rx.Observable

class GetIntroMessage(postExecutorThread: PostExecutorThread, private val repository: DatabaseRepository) : UseCase<IntroMessage>(postExecutorThread) {

    override fun buildUseCaseObservable(): Observable<IntroMessage> {
        return repository.getIntroMessage()
    }
}
